### PR TITLE
[hotfix][hive][doc] refine Hive related documentations

### PR DESCRIPTION
**Pull request by bowenli86**
## What is the purpose of the change

refine Hive related documentations

## Brief change log

- fixed some typos and wordage
- added two more data type limitations

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (docs)

### CODE REVIEW

**Comment by bowenli86** [comment]
cc @xuefuz @lirui-apache @zjuwangg

File: docs/dev/table/hive/index.md [hunk 1]
```diff
@@ -259,7 +263,8 @@ The following limitations in Hive's data types impact the mapping between Flink
 * `CHAR(p)` has a maximum length of 255
 * `VARCHAR(p)` has a maximum length of 65535
 * Hive's `MAP` only supports primitive key types while Flink's `MAP` can be any data type
+* Hive's `UNION` type is not supported
 * Flink's `INTERVAL` type cannot be mapped to Hive `INTERVAL` type
-* Flink's `TIMESTAMP_WITH_TIME_ZONE` is not supported by Hive
-* Flink's `TIMESTAMP_WITH_LOCAL_TIME_ZONE` is not supported by Hive
+* Flink's `TIMESTAMP_WITH_TIME_ZONE` and `TIMESTAMP_WITH_LOCAL_TIME_ZONE` is not supported by Hive
+* Flink's `TIMESTAMP_WITHOUT_TIME_ZONE` type cannot be mapped to Hive's `TIMESTAMP` type due to precision difference.
```
Comment (lirui-apache) [inline docs/dev/table/hive/index.md]: Also remove the type mapping from above table?

File: docs/dev/table/hive/index.md [hunk 2]
```diff
@@ -259,7 +263,8 @@ The following limitations in Hive's data types impact the mapping between Flink
 * `CHAR(p)` has a maximum length of 255
 * `VARCHAR(p)` has a maximum length of 65535
 * Hive's `MAP` only supports primitive key types while Flink's `MAP` can be any data type
+* Hive's `UNION` type is not supported
 * Flink's `INTERVAL` type cannot be mapped to Hive `INTERVAL` type
-* Flink's `TIMESTAMP_WITH_TIME_ZONE` is not supported by Hive
-* Flink's `TIMESTAMP_WITH_LOCAL_TIME_ZONE` is not supported by Hive
+* Flink's `TIMESTAMP_WITH_TIME_ZONE` and `TIMESTAMP_WITH_LOCAL_TIME_ZONE` is not supported by Hive
```
Comment (lirui-apache) [inline docs/dev/table/hive/index.md]: nit: **are** not supported

**Comment by bowenli86** [comment]
cc @xuefuz @lirui-apache @zjuwangg

**Comment by bowenli86** [comment]
merged in master and release-1.9

**Pull request closed without merging**